examples/gc/sha256.cc

```diff
@@ -0,0 +1,319 @@
+  // }
+  // std::cout << endl;
+
+  // vector<uint8_t> data_vec(padded_data, padded_data + 64);
+
+  std::string input;
+  std::cout << "请输入:";
+  std::cin >> input;
+  std::vector<uint8_t> message(input.begin(), input.end());
+
+  // 预处理消息
+  std::vector<uint8_t> data_vec = preprocess(message);
+
+  // std::vector<uint32_t> hash_vec = {0x6a09e667, 0xbb67ae85, 0x3c6ef372,
+  //                                   0xa54ff53a, 0x510e527f, 0x9b05688c,
+  //                                   0x1f83d9ab, 0x5be0cd19};
+
+  /*
+  Initial hash value, reference:
+  https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+  Section 5.3
+  */
+  std::vector<uint8_t> byte_hash_vec = {
+      0x6a, 0x09, 0xe6, 0x67, 0xbb, 0x67, 0xae, 0x85, 0x3c, 0x6e, 0xf3,
+      0x72, 0xa5, 0x4f, 0xf5, 0x3a, 0x51, 0x0e, 0x52, 0x7f, 0x9b, 0x05,
+      0x68, 0x8c, 0x1f, 0x83, 0xd9, 0xab, 0x5b, 0xe0, 0xcd, 0x19};
+
+  // 遍历 hash_vec 并将每个 uint32_t 分解为 4 个字节存储到 byte_hash_vec 中
+  // for (uint32_t value : hash_vec) {
+  //   byte_hash_vec.push_back(value >> 24 & 0xFF);
+  //   byte_hash_vec.push_back((value >> 16) & 0xFF);
+  //   byte_hash_vec.push_back((value >> 8) & 0xFF);
+  //   byte_hash_vec.push_back((value) & 0xFF);
+  // }
+  for (auto &elem : data_vec) {
+    elem = reverse_bits(elem);
+  }
+  for (auto &elem : byte_hash_vec) {
+    elem = reverse_bits(elem);
+  }
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+
+  reverse(data_vec.begin(), data_vec.end());
+
+  reverse(byte_hash_vec.begin(), byte_hash_vec.end());
+#endif
+  // std::cout << "hash";
+  // for (int i = 0; i < 32; i++) {
+  //   bitset<8> b(byte_hash_vec[i]);
+  //   std::cout << b;
+  // }
+  // std::cout << endl;
+
+  std::vector<uint8_t> input_vec;
+
+  for (auto &elem : data_vec) {
+    input_vec.push_back(elem);
+  }
+  for (auto &elem : byte_hash_vec) {
+    input_vec.push_back(elem);
+  }
+
+  // std::vector<uint128_t> output_vec(input_vec.size() / 16);
+
+  // for (size_t i = 0; i < output_vec.size(); ++i) {
+  //   uint128_t value = 0;
+  //   for (size_t j = 0; j < 16; ++j) {
+  //     value |= static_cast<uint128_t>(input_vec[i * 16 + j]) << (j * 8);
+  //   }
+  //   output_vec[i] = value;
+  // }
+
+  // for (auto &elem : output_vec) {
+  //   bitset<128> b(elem);
+  //   std::cout << b;
+  // }
+  // std::cout << endl;
+
+  std::vector<uint8_t> result(32);
+  std::string pth_str = fmt::format("{}/yacl/io/circuit/data/sha256.txt",
+                                    std::filesystem::current_path().string());
+  yacl::PlainExecutor<uint8_t> exec;
+  // std::cout << "指针内容：" << *(uint8_t *)input_vec.data() << endl;
+
+  exec.LoadCircuitFile(pth_str);
+  exec.wires_.resize(exec.circ_->nw);
+  for (size_t i = 0; i < input_vec.size(); ++i) {
+    for (int j = 0; j < 8; ++j) {
+      exec.wires_[i * 8 + j] = (input_vec[i] >> (7 - j)) & 1;  // 逐位赋值
+    }
+  }
+
+  // exec.SetupInputs(absl::MakeSpan(input_vec));  // 拼成一个vector
+
+  exec.Exec();
+
+  exec.Finalize(absl::MakeSpan(result));
```

Comment:
Finalize will only update the first 8-bits of result (as  result is defined as `std::vector<uint8_t>`, and the output number of sha256 circuit = 1), so we get the wrong output ! (i.e. only first 8 bits of output is correct)